unity-client/Assets/Scripts/MainScripts/DCL/Helpers/NFTHelper/NFTMarkets/OpenSea_Internal/Request.cs

```diff
@@ -141,61 +141,70 @@ IEnumerator Fetch(float delayRequest)
+                yield return asyncOp;
 
-                    if (!request.isNetworkError && !request.isHttpError)
-                    {
-                        response = Utils.FromJsonWithNulls<AssetsResponse>(request.downloadHandler.text);
-                    }
+                AssetsResponse response = null;
+                if (asyncOp.isSucceded)
+                {
+                    response = Utils.FromJsonWithNulls<AssetsResponse>(asyncOp.webRequest.downloadHandler.text);
+                }
 
-                    if (OpenSeaRequestController.VERBOSE)
-                        Debug.Log($"RequestGroup: Request resolving {response != null} {request.error} {url}");
-                    if (response != null)
+                if (OpenSeaRequestController.VERBOSE)
+                    Debug.Log($"RequestGroup: Request resolving {response != null} {asyncOp.webRequest.error} {url}");
+
+                if (response != null)
+                {
+                    shouldRetry = false;
+                    //if we have one element in the group, is the one failing and we dont group it again
+                    if (response.assets.Length != 0 || requests.Count <= 1)
                     {
-                        shouldRetry = false;
-                        //if we have one element in the group, is the one failing and we dont group it again
-                        if (response.assets.Length != 0 || requests.Count <= 1)
+                        using (var iterator = requests.GetEnumerator())
                         {
-                            using (var iterator = requests.GetEnumerator())
+                            while (iterator.MoveNext())
                             {
-                                while (iterator.MoveNext())
-                                {
-                                    iterator.Current.Value.Resolve(response);
-                                }
+                                iterator.Current.Value.Resolve(response);
                             }
                         }
-                        else
-                        {
-                            //There are invalids NFTs to fetch, we split the request into 2 smaller groups to find the ofender
-                            SplitGroup();
-                        }
                     }
                     else
                     {
-                        shouldRetry = retryCount < REQUEST_RETRIES;
-                        if (!shouldRetry)
+                        //There are invalids NFTs to fetch, we split the request into 2 smaller groups to find the ofender
+                        SplitGroup();
+                    }
+
+                    asyncOp.Dispose();
+                }
+                else
+                {
+                    shouldRetry = retryCount < REQUEST_RETRIES;
+                    if (!shouldRetry)
+                    {
```

Comment:
now this is handle automatically by the web request right? This is doing double retry tries

Reply:
No, in this specific case, as I commnent above of this line, we are letting this class to implement the retries.
https://github.com/decentraland/explorer/blob/cbf1c53fe9f2b43bf96d06c84c6b23f18e3dda1b/unity-client/Assets/Scripts/MainScripts/DCL/Helpers/NFTHelper/NFTMarkets/OpenSea_Internal/Request.cs#L149

As it is implementing a very specific retries system (including delays), we use our custom WebRequest system without retries (`requestAttemps: 1`) and let the current code to apply the retries.